File: ChainOfResponsibility/ChainOfResponsibilityShop/ChainOfResponsibilityShop/BronzeHandler.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ChainOfResponsibilityShop
{
    public class BronzeHandler : BaseDiscountHandler
    {
        public override void GetDiscount(string clientType)
        {
            if (clientType == "bronze")
            {
                Console.WriteLine("Bronze client gets 5% discount.");
            }
            else 
            {
            Next(clientType);
            }
        }
    }
}

File: ChainOfResponsibility/JavaExample/ChainOfResponsibility/src/BronzeHandler.java
public class BronzeHandler extends BaseDiscountHandler{
    @Override
    public void getDiscount(String clientType) {
        if ("bronze".equalsIgnoreCase(clientType)) {
            System.out.println("Bronze client gets 5% discount.");
        } else {
            next(clientType);
        }
    }
}
